unified_planning/engines/compilers/utils.py

```diff
@@ -143,11 +139,9 @@ def create_action_with_given_subs(
             get_fresh_name(problem, old_action.name, naming_list)
         )
         for p in old_action.preconditions:
-            new_action.add_precondition(substituter.substitute(p, c_subs))
+            new_action.add_precondition(p.substitute(subs))
```

Comment:
I don't know why the typing of c_subs is not accepted in the substitute method. It gives this error:
`unified_planning/engines/compilers/utils.py:145: error: Argument 4 to "create_effect_with_given_subs" has incompatible type "Dict[Parameter, FNode]"; expected "Dict[Union[FNode, Fluent, Object, Parameter, Variable, Timing, bool, int, float, Fraction], Union[FNode, Fluent, Object, Parameter, Variable, Timing, bool, int, float, Fraction]]"`

To me this does not make sense because `Parameters` is in the expected key values, but using 2 different typing is the solution I found.

Reply:
Is it still needed `c_subs` then?

Reply:
Yess